.gitignore

```diff
@@ -16,3 +16,10 @@ profiles.clj
 dev/user.clj
 .lein-repl-history
 ext/packaging
+Gemfile.lock
+Gemfile.local
+.DS_Store
+.idea/
+/junit/
+/tmp/
+puppet_server.iml
```

Comment:
I think we've generally tried to keep IDE-specific lines out of the .gitignore - stuff like .idea and puppet_server.iml.  Instead, we've been assuming developers would put the appropriate IDE-specific references in their global .gitignores.

Also, are /junit/ and /tmp/ needed?  I can't remember what .DS_Store is either...